### PR TITLE
RDM-13140 FTAs for the getLinkedCases API 

### PR DESCRIPTION
**Pull request by mattnayler**
### JIRA link (if applicable) ###

   [RDM-13140](https://tools.hmcts.net/jira/browse/RDM-13140) _"Functional Tests for the getLinkedCases API"_
   [RDM-13139](https://tools.hmcts.net/jira/browse/RDM-13139) _"getLinkedCases API - retrieve case links from database and return response payload"_

### Change description ###

Functional tests for the new getLinkedCases API as described in RDM-13139, see #1936 & #1960.

**Does this PR introduce a breaking change?** (check one with "x")

This PR is dependent on hmcts/ccd-definition-store-api#1178 and as a result it contains the updates from #1916. 

This PR uses the test defintions from hmcts/ccd-test-definitions#47 

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: Jenkinsfile_CNP [hunk 1]
```diff
@@ -19,7 +19,8 @@ def branchesToSync = ['demo', 'ithc', 'perftest']
 
 // Variables to switch pipeline logic and wiring per type of build
 
-def definitionStoreDevelopPr = "PR-575"  // This doesn't change frequently, but when it does, only change this value.
+// TODO: remove TEMPORARY OVERRIDE for RDM-13140, reinstate: def definitionStoreDevelopPr = "PR-575"
+def definitionStoreDevelopPr = "PR-1212"  // This doesn't change frequently, but when it does, only change this value.
```
Comment (mattnayler) [inline Jenkinsfile_CNP:23]: Temporary change needs reverting before merger.

File: charts/ccd-data-store-api/values.preview.template.yaml [hunk 2]
```diff
@@ -27,7 +27,8 @@ java:
     DATA_STORE_DB_PASSWORD: "{{ .Values.postgresql.postgresqlPassword}}"
     DATA_STORE_DB_OPTIONS: "?stringtype=unspecified"
     DATA_STORE_DB_MAX_POOL_SIZE: 10
-    DEFINITION_STORE_HOST: http://ccd-definition-store-api-pr-575.service.core-compute-preview.internal/
+    # TODO: remove TEMPORARY OVERRIDE for RDM-13140, reinstate: DEFINITION_STORE_HOST: http://ccd-definition-store-api-pr-575.service.core-compute-preview.internal/
+    DEFINITION_STORE_HOST: http://ccd-definition-store-api-pr-1212.service.core-compute-preview.internal/
```
Comment (mattnayler) [inline charts/ccd-data-store-api/values.preview.template.yaml:31]: Temporary change needs reverting before merger.

File: src/aat/java/uk/gov/hmcts/ccd/datastore/befta/DataStoreTestAutomationAdapter.java [hunk 3]
```diff
@@ -182,6 +182,8 @@ public Object calculateCustomValue(BackEndFunctionalTestScenarioContext scenario
             }
         } else if (key.toString().equalsIgnoreCase("dateTwentyDaysFromToday")) {
             return LocalDate.now().plusDays(20).toString();
+        } else if (key.toString().equalsIgnoreCase("generateUUID")) {
```
Comment (sullivangareth) [inline src/aat/java/uk/gov/hmcts/ccd/datastore/befta/DataStoreTestAutomationAdapter.java:185]: Not part of this change, but this method needs some serious refactoring !

File: src/aat/resources/features/F-142 - Get Linked Case V2 External/CreateCases/F-142_CreateLinkedCase_NonStandardCaseLink.td.json [hunk 4]
```diff
@@ -0,0 +1,38 @@
+{
+  "_guid_": "F-142_CreateLinkedCase_NonStandardCaseLink",
+  "_extends_": "F-142_CreateCase_MasterCaseType__Base",
+
+  "request": {
+    "body": {
+      "data": {
+        "caseNameHmctsInternal": "Name internal: ${[scenarioContext][customValues][generateUUID]}",
```
Comment (sullivangareth) [inline src/aat/resources/features/F-142 - Get Linked Case V2 External/CreateCases/F-142_CreateLinkedCase_NonStandardCaseLink.td.json:8]: Does this need to be a UUID here given that we only check its a non null string in the response ?

File: src/aat/resources/features/F-142 - Get Linked Case V2 External/CreateCases/F-142_CreateLinkedCase_StandardCaseLink.td.json [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+  "_extends_": "F-142_CreateCase_MasterCaseType__Base",
+
+  "request": {
+    "body": {
+      "data": {
+        "caseNameHmctsInternal": "Name internal: ${[scenarioContext][customValues][generateUUID]}",
+
+        "caseLinks": [
+          {
+            "id": "52837798-42c6-43cc-98f6-0895fdba4961",
```
Comment (sullivangareth) [inline src/aat/resources/features/F-142 - Get Linked Case V2 External/CreateCases/F-142_CreateLinkedCase_StandardCaseLink.td.json:12]: Can this not use `${[scenarioContext][customValues][generateUUID]}` to generate a UUID rather than hardcoding it ?

Comment (sullivangareth) [reply src/aat/resources/features/F-142 - Get Linked Case V2 External/CreateCases/F-142_CreateLinkedCase_StandardCaseLink.td.json:12]: Or is it because we cant reference that value when asserting the id in the response ?

**Review by sullivangareth — APPROVED** [review]

File: src/aat/java/uk/gov/hmcts/ccd/datastore/befta/DataStoreTestAutomationAdapter.java (hunk 3 above)
Comment (mattnayler) [reply src/aat/java/uk/gov/hmcts/ccd/datastore/befta/DataStoreTestAutomationAdapter.java:185]: agreed.

File: src/aat/resources/features/F-142 - Get Linked Case V2 External/CreateCases/F-142_CreateLinkedCase_StandardCaseLink.td.json (hunk 5 above)
Comment (mattnayler) [reply src/aat/resources/features/F-142 - Get Linked Case V2 External/CreateCases/F-142_CreateLinkedCase_StandardCaseLink.td.json:12]: The latter. Although it would only become an issue in those matching TD files that create cases with mulitple links or multipe reasons for a link as I couldn't use `[[ANYTHING_PRESENT]]`; I would need the ID to be present for the `expectedResponse` assertion.

File: src/aat/resources/features/F-142 - Get Linked Case V2 External/S-142.13.td.json [hunk 6]
```diff
@@ -0,0 +1,174 @@
+{
+  "_guid_": "S-142.13",
+  "_extends_": "Get_Linked_Cases__Base",
+
+  "title": "Continue from AC05, startRecordNumber = 6, maxReturnRecordCount = 5, return 200 with 4 case links in response payload",
```
Comment (mattnayler) [inline src/aat/resources/features/F-142 - Get Linked Case V2 External/S-142.13.td.json:5]: this file implies this case has 5 case liks in the response payload

Comment (mattnayler) [reply src/aat/resources/features/F-142 - Get Linked Case V2 External/S-142.13.td.json]: Adjusted in eb41fe1d6c82bfa48c50852f77bbdbcc0bcaa7d1

File: build.gradle [hunk 7]
```diff
@@ -271,7 +271,7 @@ dependencies {
     compile 'org.jooq:jool-java-8:0.9.14'
     compile 'com.github.hmcts:ccd-case-document-am-client:1.7.1'
 
-    testCompile group: 'com.github.hmcts', name: 'ccd-test-definitions', version: '7.18.0'
+    testCompile group: 'com.github.hmcts', name: 'ccd-test-definitions', version: '7.18.1-prerelease'
```
Comment (mattnayler) [inline build.gradle:274]: Needs updating after release of hmcts/ccd-test-definitions#47.
